File: zodiac_art/renderer/glyphs.py
"""Glyph mappings for zodiac signs and planets."""

from __future__ import annotations

from zodiac_art.config import load_config


ZODIAC_GLYPHS_UNICODE: dict[str, str] = {
    "Aries": "♈",
    "Taurus": "♉",
    "Gemini": "♊",
    "Cancer": "♋",
    "Leo": "♌",
    "Virgo": "♍",
    "Libra": "♎",
    "Scorpio": "♏",
    "Sagittarius": "♐",
    "Capricorn": "♑",
    "Aquarius": "♒",
    "Pisces": "♓",
}

PLANET_GLYPHS_UNICODE: dict[str, str] = {
    "Sun": "☉",
    "Moon": "☾",
    "Mercury": "☿",
    "Venus": "♀",
    "Mars": "♂",
    "Jupiter": "♃",
    "Saturn": "♄",
    "Uranus": "♅",
    "Neptune": "♆",
    "Pluto": "♇",
}

ZODIAC_GLYPHS_ASCII = {key: key.upper() for key in ZODIAC_GLYPHS_UNICODE}
PLANET_GLYPHS_ASCII = {key: key.upper() for key in PLANET_GLYPHS_UNICODE}


def get_zodiac_glyph(sign: str) -> str:
    """Return a glyph for the sign based on config."""

    config = load_config()
    if config.glyph_mode == "ascii":
        return ZODIAC_GLYPHS_ASCII.get(sign, sign.upper())
    return ZODIAC_GLYPHS_UNICODE.get(sign, sign)


def get_planet_glyph(planet: str) -> str:
    """Return a glyph for the planet based on config."""

    config = load_config()
    if config.glyph_mode == "ascii":
        return PLANET_GLYPHS_ASCII.get(planet, planet.upper())
    return PLANET_GLYPHS_UNICODE.get(planet, planet)

File: editor/src/utils/glyphs.ts
const PLANET_GLYPHS: Record<string, string> = {
  sun: '☉',
  moon: '☾',
  mercury: '☿',
  venus: '♀',
  mars: '♂',
  jupiter: '♃',
  saturn: '♄',
  uranus: '♅',
  neptune: '♆',
  pluto: '♇',
}

const SIGN_GLYPHS: Record<string, string> = {
  aries: '♈',
  taurus: '♉',
  gemini: '♊',
  cancer: '♋',
  leo: '♌',
  virgo: '♍',
  libra: '♎',
  scorpio: '♏',
  sagittarius: '♐',
  capricorn: '♑',
  aquarius: '♒',
  pisces: '♓',
}

export function formatSelectionLabel(id: string): string {
  if (id === 'chartRoot') {
    return 'Chart'
  }
  if (id === 'chart.background') {
    return 'Chart background'
  }
  if (id === 'chart.background_image') {
    return 'Background image'
  }
  if (id === 'asc.marker') {
    return 'Ascendant marker ↑'
  }
  if (id.startsWith('planet.')) {
    const parts = id.split('.')
    const name = parts[1] || 'Planet'
    const subtype = parts[2]
    const glyph = PLANET_GLYPHS[name.toLowerCase()]
    if (subtype === 'glyph') {
      return glyph ? `Planet ${name} glyph ${glyph}` : `Planet ${name} glyph`
    }
    if (subtype === 'label') {
      return `Planet ${name} label`
    }
    return glyph ? `Planet ${name} ${glyph}` : `Planet ${name}`
  }
  if (id.startsWith('sign.')) {
    const parts = id.split('.')
    const name = parts[1] || 'Sign'
    const glyph = SIGN_GLYPHS[name.toLowerCase()]
    return glyph ? `Sign ${name} ${glyph}` : `Sign ${name}`
  }
  return id
}
